3.5.0 and new strict mode: java.lang.IllegalStateException: No entity type for ...

Hi Timo,

I just tried 3.5.0 as you fixed a couple of bugs a colleague of mine reported (#908, #909, #910).

I hit a problem with the processor phase due to the new strictMode behavior: we have a class which isn't annotated with a JPA annotation in our hierarchy and we now have:
[ERROR] error on execute: java.lang.IllegalStateException: No entity type for fr.openwide.core.jpa.business.generic.model.GenericEntity
which seems to be normal due to this commit: https://github.com/querydsl/querydsl/commit/3874f72ee01fc3482513cf758c41f054b2de84b0#diff-04ba2bdca9a881f50dd2d88b4fa35810

2 questions:
1/ are you sure it's a good idea to trigger an error in this case? AFAICS we don't have any issue with this configuration using Hibernate in JPA2 mode and it seems to be supported: http://docs.oracle.com/javaee/6/tutorial/doc/bnbqn.html#bnbqq
2/ I don't see a way to configure the strictMode via the com.mysema.query.apt.jpa.JPAAnnotationProcessor: AFAICS, it's not designed to be configured at all. Is it your intention.

Thanks for your feedback!

--- a/querydsl-codegen/src/main/java/com/mysema/query/codegen/EntitySerializer.java
+++ b/querydsl-codegen/src/main/java/com/mysema/query/codegen/EntitySerializer.java
@@ -220,9 +220,12 @@ public class EntitySerializer implements Serializer {
     protected void initEntityFields(CodeWriter writer, SerializerConfig config,
             EntityType model) throws IOException {
         Supertype superType = model.getSuperType();
-        if (superType != null && superType.getEntityType() != null && superType.getEntityType().hasEntityFields()) {
-            Type superQueryType = typeMappings.getPathType(superType.getEntityType(), model, false);
-            writer.line("this._super = new " + writer.getRawName(superQueryType) + "(type, metadata, inits);");
+        if (superType != null) {
+            EntityType entityType = superType.getEntityType();
+            if (entityType != null && entityType.hasEntityFields()) {
+                Type superQueryType = typeMappings.getPathType(entityType, model, false);
+                writer.line("this._super = new " + writer.getRawName(superQueryType) + "(type, metadata, inits);");
+            }
         }
 
         for (Property field : model.getProperties()) {

